modules/instrument_builder/jsx/react.questions.js

```diff
@@ -0,0 +1,462 @@
+			                </li>
+			                <li onClick={this.selectType.bind(this, "numeric", "Numeric")}>
+			                    <a id="numeric" className="option" title="User data entry of a number">Numeric</a>
+			                </li>
+			                <li className="divider"></li>
+			                <li>
+			                    <div className="col-sm-12"><h5 className="">Formatting</h5></div>
+			                </li>
+			                <li onClick={this.selectType.bind(this, "line", "Blank Line")}>
+			                    <a id="line" className="option" title="Empty line">Blank Line</a>
+			                </li>
+			                <li onClick={this.selectType.bind(this, "page-break", "Page Break")}>
+			                    <a id="page-break" className="option" title="Start a new page">Page Break</a>
+			                </li>
+			            </ul>
+	                </div>
+	            </div>
+	        </div>
+		)
+	}
+});
+
+AddElement = React.createClass({
+	getInitialState: function() {
+		var state;
+		if(this.props.element){
+			state = {
+				Options: this.props.element.Options,
+		 		Description: this.props.element.Description,
+		 		Name: this.props.element.Name,
+		 		selected: this.props.element.selected
+			}
+		} else {
+			state = {
+				Options: {},
+		 		Description: '',
+		 		Name: '',
+		 		selected: {
+		 			id: '',
+		 			value: 'Select One'
+		 		}
+			}
+		}
+	 	return state;
+	},
+	updateState: function(newState) {
+		this.setState(newState);
+	},
+	addQuestion: function () {
+		var selected = this.state.selected.id,
+			questionText = this.state.Description,
+			questionName = this.state.Name,
+			hasError = false,
+			element;
+	    if(!selected) {
+	        alert("No element type selected");
+	        return;
+	    }
+
+	    if(questionText == '' && selected != 'line') {
+	        if(selected == 'page-break') {
+	        	this.setState(function(state){
+	        		var temp = (state.error) ? state.error : {};
+	        		temp.questionText = "Must use question text as page header";
+					return {
+						error: temp
+					};
+				});
+				hasError = true;
+	        } else {
+	        	this.setState(function(state){
+	        		var temp = (state.error) ? state.error : {};
+	        		temp.questionText = "No question text specified";
+					return {
+						error: temp
+					};
+				});
+				hasError = true;
+	        }
+	    }
+	    if (!hasError && this.state.error) {
+	    	delete this.state.error.questionText;
+	    }
+	    if(questionName == '' && selected != "header" && selected != "label" && selected != 'line' && selected != 'page-break') {
+	    	this.setState(function(state){
+        		var temp = (state.error) ? state.error : {};
+        		temp.questionName = "Must specifiy name for database to save value into";
+				return {
+					error: temp
+				};
+			});
+	        hasError = true;
+	    } else if (this.state.error) {
+	    	delete this.state.error.questionName;
```

Comment:
You should only modify state through calls to setState (or replaceState), and not mutate the state object directly, since changing it directly won't trigger the appropriate rendering calls.